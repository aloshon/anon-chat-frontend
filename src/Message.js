const Message = ({message, timestamp}) => {
    return (
        <div>
            <p style={{marginBottom: 0}}>{message}</p>
            <small>{timestamp}</small>
        </div>
    )
}

export default Message;