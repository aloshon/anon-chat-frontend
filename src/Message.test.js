import { render, screen } from '@testing-library/react';
import Message from './Message';


describe("Message should display text and timestamp of when the message was created", () => {
    const timestamp = "Sun, 25 Apr 2021 21:39:55 GMT";
    const message = "this is a test";

    const {asFragment} = render(<Message message={message} timestamp={timestamp}/>);

    test("renders message with text and timestamp", () => {
      const message = screen.getByText("this is a test");
      const timestampYear = screen.getByText("Sun, 25 Apr 2021 21:39:55 GMT");
      expect(message).toBeInTheDocument();
      expect(timestampYear).toBeInTheDocument();
    });
  
    test("matches snapshot", () => {
      expect(asFragment()).toMatchSnapshot();
    });
    
  });