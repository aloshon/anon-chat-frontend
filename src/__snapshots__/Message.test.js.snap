// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message should display text and timestamp of when the message was created matches snapshot 1`] = `<DocumentFragment />`;
